### ISSUE TITLE
Integration with spacy !

### ISSUE THREAD

**Issue by osoblanco** [aimhubio/aim#1351]
## 🚀 Feature

I recently caught myself rummaging through a ton of logs after using spacy. It would be very nice if internally I had the option to track everything with aim.

### Motivation

The default logger is not particularly user/filtering friendly, thus it would be nice if I would be able to track live with something more visual and easier to search around. Also, it would be a great addition if my experiments and displacy visualisations were in one tab, which will be a quality of life improvement.

### Pitch

Allow me to go from this.

```

=========================== Initializing pipeline ===========================
[2021-11-12 12:02:52,862] [INFO] Set up nlp object from config
[2021-11-12 12:02:52,870] [INFO] Pipeline: ['sentencizer', 'tok2vec', 'ner']
[2021-11-12 12:02:53,145] [INFO] Created vocabulary
[2021-11-12 12:02:53,146] [INFO] Finished initializing nlp object
[2021-11-12 12:02:58,148] [INFO] Initialized pipeline components: ['tok2vec', 'ner']
✔ Initialized pipeline

============================= Training pipeline =============================
ℹ Pipeline: ['sentencizer', 'tok2vec', 'ner']
ℹ Frozen components: ['sentencizer']
ℹ Initial learn rate: 0.001

...
```

To this

![Screenshot from 2022-02-16 14-29-23](https://user-images.githubusercontent.com/8036160/154246134-e7408452-95a6-46a1-ae2a-ccbc53a7f207.png)

**Comment by gorarakelyan** [comment]
hey @osoblanco,

will defo prioritize spacy integration in the next couple of weeks! this is one of the highly requested features

**Comment by AmirF194** [comment]
I put together a spaCy training logger for Aim: `aim.AimLogger.v1`, registered through `spacy.util.registry.loggers` the same way spacy-loggers' WandbLogger and MLflowLogger are, so it plugs into `[training.logger]` in a training config. It creates an `aim.Run` and tracks `score`/`other_scores` per step plus per-pipe `loss` (context `{"pipeline": name}`), using the same `run.track()` calls `aim.lightgbm` and `aim.keras` already use.

Ran it end to end against spaCy 3.7.5 and current `aim` main, no mocks: trained a tiny blank-en textcat pipeline for 5 epochs with the logger wired in through the registry, exactly as `spacy train` would call it. Loss fell each epoch (0.77 -> 0.05), score and cats_macro_f logged too. Reopened the `.aim` repo in a fresh process afterward and read the values back through `Repo.get_run()`: loss/score/cats_macro_f all present, 5 points each, matching the training output.

One open gap: a cold `Repo.get_run()` only sees the run's data after something builds the index (I used `aim up` briefly); `aim runs ls` finds the run without that step. That looked like a storage-layer thing shared by every adapter, not specific to this one, so flagging it rather than treating it as this logger's bug.

Open question: should this live in aim's own tree as `aim.spacy` (same shape as `aim.lightgbm`/`aim.keras`), or is spacy-loggers (explosion's package, where wandb/mlflow/comet already live) the better home for it?

Can open the PR either way.
